components/ibm-components/watson/deploy/src/wml-deploy.py

```diff
@@ -0,0 +1,99 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# 
+# define the function to deploy the model
+
+def deploy(args):
+    from watson_machine_learning_client import WatsonMachineLearningAPIClient
+    import boto3
+    import os
+    
+    wml_model_name = args.model_name
+    wml_scoring_payload = args.scoring_payload
+    model_uid = args.model_uid
+
+    # retrieve credentials
+    with open("/app/secrets/wml_url", 'r') as f:
+        wml_url = f.readline().strip('\'')
+    f.close()
+    with open("/app/secrets/wml_username", 'r') as f:
+        wml_username = f.readline().strip('\'')
+    f.close()
+    with open("/app/secrets/wml_password", 'r') as f:
+        wml_password = f.readline().strip('\'')
+    f.close()
+    with open("/app/secrets/wml_instance_id", 'r') as f:
+        wml_instance_id = f.readline().strip('\'')
+    f.close()
+
+    with open("/app/secrets/s3_endpoint", 'r') as f:
+        s3_endpoint = f.readline().strip('\'')
+    f.close()
+    with open("/app/secrets/s3_access_key", 'r') as f:
+        s3_access_key = f.readline().strip('\'')
+    f.close()
+    with open("/app/secrets/s3_secret_key", 'r') as f:
+        s3_secret_key = f.readline().strip('\'')
+    f.close()
+
+    with open("/app/secrets/s3_input_bucket", 'r') as f:
+        s3_input_bucket = f.readline().strip('\'')
+    f.close()
+
+    # set up the WML client
+    wml_credentials = {
+                       "url": wml_url,
+                       "username": wml_username,
+                       "password": wml_password,
+                       "instance_id": wml_instance_id
+                      }
+    client = WatsonMachineLearningAPIClient( wml_credentials )
+        
+    # deploy the model
+    deployment_name    = wml_model_name
+    deployment_desc    = "deployment of %s" %wml_model_name
+    deployment       = client.deployments.create( model_uid, deployment_name, deployment_desc )
+    scoring_endpoint = client.deployments.get_scoring_url( deployment )
+    print( "scoring_endpoint: ", scoring_endpoint )
+
+    # download scoring payload
+    payload_file = os.path.join('/app', wml_scoring_payload)
+    
+    s3 = boto3.resource('s3',
```

Comment:
@vicaire this is something we can definitely come back to in future, as we strive to take the message of 'Watson anywhere' forward. Currently s3 client is something most of the developer use

Reply:
Hi animeshsingh@, we are moving towards making all the components checked into the repository compatible with multiple clouds. 

Minio Client should be simple to use (as much as using an S3 client directly) and would make the component work across multiple clouds. 

It would also make it easy to add a end-to-end test for this component (our e2e test framework uses GCP). e2e tests are important as they automatically verify that the component keeps working at each PR commit.

Reply:
Thanks @vicaire . We will look into minio client. Vis a vis the end to end test, how are we doing it for components which use GCP, for e.g. Cloud Dataflow service? Do we create a dedicated set of credentials? If not, the tests are mostly mocking the behaviour?

Reply:
@vicaire We have switched to minio client.  End to end tests - want to get your input as mentioned above